odoo/addons/base/models/res_company.py

```diff
@@ -393,3 +394,15 @@ def _all_branches_selected(self):
             len(self.env.companies.root_id) == 1
             and self.env.companies == self.env['res.company'].sudo().search([('id', 'child_of', self.env.company.root_id.ids)])
         )
+
+    def action_res_company_branches_list(self):
+        self.ensure_one()
+        view_id = self.env.ref('base.view_company_branches_tree').id
+        return {
+            'type': 'ir.actions.act_window',
+            'name': _('Branches'),
+            'res_model': 'res.company',
+            'domain': [('parent_id', '=', self.id)],
+            'context': {'default_parent_id': self.id},
```

Comment:
Wouldn't it be clearer to have `active_test=False` in the action's context (this one) instead of the context of the button?  If you ever do so, I suggest to rename the method `action_all_company_branches`.

Reply:
Perfectly fine. I moved it to the action and updated the name.